test/Proxy/ImageDownloaderTest.java

```diff
@@ -0,0 +1,30 @@
+
+package Proxy;
+
+import Proxy.Classes.Image;
+import Proxy.Classes.UserProfile;
+import Proxy.cached.CachedImageProvider;
+import Proxy.cached.ImageProvider;
+import java.util.HashMap;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class ImageDownloaderTest {
+    
+    public ImageDownloaderTest() {
+    }
+
+    @Test
+    public void testGetPopularImageValidated() {
```

Comment:
I think we can have more unit tests, I'd recommend practicing TDD to achieve coverage by default.